src/ol/renderer/Map.js

```diff
@@ -105,14 +106,19 @@ class MapRenderer extends Disposable {
     let result;
     const viewState = frameState.viewState;
     const viewResolution = viewState.resolution;
+    const managedLayers = frameState.layerStatesArray.map(function(state) {
+      if (state.managed) {
+        return getUid(state.layer);
+      }
+    });
 
     /**
      * @param {import("../Feature.js").FeatureLike} feature Feature.
      * @param {import("../layer/Layer.js").default} layer Layer.
      * @return {?} Callback result.
      */
     function forEachFeatureAtCoordinate(feature, layer) {
-      const managed = frameState.layerStates[getUid(layer)].managed;
+      const managed = includes(managedLayers, getUid(layer));
```

Comment:
We could retain the previous performance by making a managed layers lookup object instead of an array.  (I haven’t confirmed this would be a performance improvement, but I’m guessing the property check would be cheaper than the includes call.)

Reply:
See 5d1c27d
Instead of creating a structure for the managed layers, the value is now directly bounded to the callback function.